libnd4j/include/ops/declarable/helpers/cpu/lup.cpp

```diff
@@ -237,25 +237,65 @@ namespace helpers {
         samediff::Threads::parallel_tad(loop, currentRow + 1, rowNum, 1);
     }
 
+    template <typename T>
+    static void doolitleLU(LaunchContext* context, NDArray* compound, Nd4jLong rowNum) {
+        auto input = compound->dup();
```

Comment:
Can we avoid this?

Reply:
lu procedure only inplace, so it's inevitable.